dora/core/server/worker/src/test/java/alluxio/worker/dora/PagedFileReaderTest.java

```diff
@@ -0,0 +1,196 @@
+    Path path = Paths.get(localUfsRoot, "testFile");
+    mTestData = BufferUtils.getIncreasingByteArray(mFileLen);
+    try (FileOutputStream os = new FileOutputStream(path.toFile())) {
+      os.write(mTestData);
+    }
+    mTestFileName = path.toString();
+    CacheManagerOptions cacheManagerOptions = CacheManagerOptions
+        .createForWorker(mConf);
+    String fileId = new AlluxioURI(mTestFileName).hash();
+    CacheManager cacheManager = CacheManager.Factory.create(
+        mConf, cacheManagerOptions, pageMetaStore);
+    mPagedFileReader = PagedFileReader.create(
+        mConf, cacheManager, mLocalUfs, fileId, mTestFileName, mFileLen, 0);
+    mPositionReaderTest = new PositionReaderTest(mPagedFileReader, mFileLen);
+  }
+
+  @After
+  public void after() throws IOException {
+    mPagedFileReader.close();
+    mLocalUfs.close();
+    mEmbeddedChannel.close();
+  }
+
+  @Test
+  public void testAllCornerCases() throws IOException {
+    mPositionReaderTest.testAllCornerCases();
+  }
+
+  @Test
+  public void testReadRandomPart() throws IOException {
+    mPositionReaderTest.testReadRandomPart();
+  }
+
+  @Test
+  public void testConcurrentReadRandomPart() throws Exception {
+    mPositionReaderTest.concurrentReadPart();
+  }
+
+  @Test
+  public void read() throws IOException {
+    int testNum = Math.min(mFileLen, mMinTestNum);
+    for (int i = 0; i < testNum; i++) {
+      int offset = mRandom.nextInt(mFileLen);
+      int readLength = mRandom.nextInt(mFileLen - offset);
+      byte[] tmpBytes = Files.readAllBytes(Paths.get(mTestFileName));
+      byte[] bytes = Arrays.copyOfRange(tmpBytes, offset, offset + readLength);
+      ByteBuffer realByteBuffer = ByteBuffer.wrap(bytes);
+      ByteBuffer byteBuffer = mPagedFileReader.read(offset, readLength);
+      Assert.assertEquals(realByteBuffer, byteBuffer);
+    }
+  }
+
+  @Test
+  public void transferTo() throws IOException {
+    int testNum = Math.min(mFileLen, mMinTestNum);
+    for (int i = 0; i < testNum; i++) {
+      int offset = mRandom.nextInt(mFileLen);
+      mPagedFileReader.setPosition(offset);
+      ByteBuf byteBuf = Unpooled.buffer(mFileLen);
+
+      int bytesRead = mPagedFileReader.transferTo(byteBuf);
+      byte[] realBytesArray = new byte[bytesRead];
+      System.arraycopy(mTestData, offset, realBytesArray, 0, bytesRead);
+      byte[] bytesArray = new byte[byteBuf.readableBytes()];
+      byteBuf.readBytes(bytesArray);
+      assertTrue(bytesRead > 0);
+      assertArrayEquals(realBytesArray, bytesArray);
+    }
+  }
+
+  @Test
+  public void getLength() {
+    Assert.assertEquals(mPagedFileReader.getLength(), mFileLen);
+  }
+
+  @Test
+  public void getMultipleDataFileChannel() throws IOException {
+    CompositeDataBuffer compositeDataBuffer =
+        mPagedFileReader.getMultipleDataFileChannel(mEmbeddedChannel, mFileLen);
+    Assert.assertEquals(mFileLen, compositeDataBuffer.getLength());
+  }
```

Comment:
add tests for lengths other than `mFileLen`
test the content of `compositeDataBuffer` is the same as the file in the UFS.